### PR TITLE
Add reverse proxying of /static/* assets.

### PR DESCRIPTION
**Pull request by fotinakis**
Currently our production load balancer is configured to short-circuit requests to /static/* and serve assets directly from a GCS bucket. This replicates that ability into our nginx setup directly to help enable the upcoming load balancer + deployment changes.

Note: this will not have any effect on real users until load balancing infrastructure is changed and the GCLB is removed.

Tested:

index.html requests still work and return no-cache headers:
```
$ curl -I http://dev.percy.local:8080/                
...
Cache-Control: private,no-cache,no-store,must-revalidate,max-age=0

$ curl -I http://dev.percy.local:8080/foo
...
Cache-Control: private,no-cache,no-store,must-revalidate,max-age=0
```

Old assets not even present in the app can be correctly served from /static, and have the correct headers (esp. Cache-Control):

```
$ curl -I http://dev.percy.local:8080/static/assets/vendor-e59a7e1227f7825bb9c1137ed765300c.js
HTTP/1.1 200 OK
Server: nginx/1.10.3 (Ubuntu)
Date: Tue, 05 Dec 2017 22:36:34 GMT
Content-Type: application/javascript
Connection: keep-alive
Cache-Control: public, max-age=2678400,no-transform
Expires: Fri, 05 Jan 2018 22:36:13 GMT
Last-Modified: Wed, 04 Oct 2017 01:39:04 GMT
ETag: "76b20c5456e1946c6eba146140b9c57d"
Content-Encoding: gzip
X-Frame-Options: DENY
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "eslint-check": "eslint --print-config .eslintrc.js | eslint-config-prettier-check",
     "build": "ember build",
     "build-production": "yarn run clean && ember build --environment=production",
-    "clean": "rm -rf dist/* tmp/*",
```
Comment (fotinakis) [inline package.json]: Sneaking in a fix for a tiny local bug I hit.

Comment (timhaines) [reply package.json]: Naughty

File: utils/deploy [hunk 2]
```diff
@@ -26,7 +26,7 @@ rm -rf dist/
 docker cp $CONTAINER_ID:/app/src/dist/ dist/
 
 # Upload new assets to GCS.
-gsutil -m -h "Cache-Control: public, max-age=2678400" cp \
+gsutil -m -h "Cache-Control: public,max-age=2678400" cp \
```
Comment (timhaines) [inline utils/deploy]: presumably this is intentional.

File: config/nginx-default-site.conf [hunk 3]
```diff
@@ -1,6 +1,11 @@
 set_real_ip_from 0.0.0.0/0;
 real_ip_header CF-Connecting-IP;
 
+upstream gs {
```
Comment (timhaines) [inline config/nginx-default-site.conf]: if this is an acronym for google-storage, can we call it google-storage?

File: config/nginx-default-site.conf [hunk 4]
```diff
@@ -38,7 +41,42 @@ server {
...
+    proxy_hide_header x-goog-metageneration;
+    proxy_hide_header x-goog-stored-content-encoding;
+    proxy_hide_header x-goog-stored-content-length;
+    proxy_hide_header x-goog-storage-class;
+    proxy_hide_header x-xss-protection;
+    proxy_hide_header accept-ranges;
+    proxy_hide_header Set-Cookie;
+    proxy_ignore_headers Set-Cookie;
+
+    # Cache-Control header is set on GCS objects on upload and passes through correctly.pdep
```
Comment (timhaines) [inline config/nginx-default-site.conf]: pdep is unintentional typo?

**Pull request merged**